### ISSUE TITLE
build 2.0 branch on kwin git

### ISSUE THREAD

**Issue by alexjp** [a-parhom/LightlyShaders#80]
It seems on kde-git/kwin git version the effect doesnt build.

I was able to make it build here:
By adding to CMakeLists.txt `set(CMAKE_CXX_STANDARD 20)`

and by changing `WindowBlurBehindRole` to `ScriptedEffect::DataRole::WindowBlurBehindRole` on line 469 and 491 of `lightlyshaders.cpp`

Thanks for the great kwin effect!

**Comment by alexjp** [comment]
well ... currently, there is this breaking change that makes it not work on git master:

https://invent.kde.org/plasma/kwin/-/merge_requests/3189

**Comment by luisbocanegra** [comment]
> well ... currently, there is this breaking change that makes it not work on git master:
> 
> [invent.kde.org/plasma/kwin/-/merge_requests/3189](https://invent.kde.org/plasma/kwin/-/merge_requests/3189)

So finally happened... @a-parhom mentioned that in https://github.com/a-parhom/LightlyShaders/issues/65 hopefully he ports it to OffscreenEffect

Also, this is the other change https://invent.kde.org/plasma/kwin/-/merge_requests/2982 that caused the WindowBlurBehindRole error

**Comment by a-parhom** [comment]
I've tried porting this effect to the OffscreenEffect. That works somehow, but is not a very good solution even for the Plasma's builtin Invert effect, which I've reported to the devs. Hopefully, they will fix that for the upcoming Plasma 5.27.

Another problem is the API change related to setting window blur region. The workaround, that I've found for the 5.27, works only for X11. The devs told me they will think about providing new API for setting blur regions, but I'm not sure it will be ready till 5.27.
